src/lens.jl

```diff
@@ -173,6 +179,8 @@ end
 struct IndexLens{I <: Tuple} <: Lens
     indices::I
 end
+==(l1::IndexLens{I}, l2::IndexLens{I}) where {I} = l1.indices == l2.indices
+hash(l::IndexLens, h::UInt) = hash(l.indices, h)
```

Comment:
One minor inconvenience with this is that `hash(@lens _[index]) == hash(index)` holds, e.g. hash forgets about the type. A possible and performant fix would be:
```julia
julia> objectid(IndexLens) # inspiration for the salt
0x8aa710fd8cb95e7e
```
```julia
function hash(l::IndexLens, h::UInt)
    salt = 0x8aa710fd8cb95e7e
    hash(l.indices, hash(salt, h))
end
```
And similar for `ComposedLens`. If there is a performance hit for doing this, we can also leave as is.